client/src/rr/dnssec/signer.rs

```diff
@@ -240,6 +240,7 @@ use serialize::binary::BinEncoder;
 ///    NSEC RRs needed to authenticate the response (see Section 3.1.3).
 /// ```
 #[cfg(any(feature = "openssl", feature = "ring"))]
+#[derive(Clone)]
```

Comment:
I don't think the functionality of this PR, making things `Send` and `Sync`, should involve cloning Signer, which could be expensive and maybe not even possible (e.g. if/when you support PKCS#11, you simply might not be able to have `Signer` implement `Clone`). Instead I'd try to get `Signer` to be `Sync` and then change all the things holding `Signer` to `Rc<Signer>`.

Reply:
yeah, I was thinking about this. It would have to be `Arc`, but that would work. I'll see about doing that before merging this. That's great feedback.